src/test/java/com/epam/tc/hw2/ex1/FirstExercise.java

```diff
@@ -0,0 +1,135 @@
+
+        //2. Assert Browser title
+        String actualTitle = driver.getTitle();
+        softly.assertThat(actualTitle).isEqualTo(EXPECTED_TITLE);
+
+        //3. Perform login
+        WebElement signIn = driver.findElement(SIGN_IN_BUTTON);
+        signIn.click();
+
+        WebElement login = (new WebDriverWait(driver, 5))
+                .until(ExpectedConditions.visibilityOfElementLocated(LOGIN_INPUT_BUTTON));
+        login.sendKeys(LOGIN_DATA);
+
+        WebElement password = driver.findElement(PASSWORD_INPUT_BUTTON);
+        password.sendKeys(PASSWORD_DATA);
+
+        WebElement enter = driver.findElement(ENTER_BUTTON);
+        enter.click();
+
+        //4. Assert Username is logged in
+        String actualData = driver.findElement(USER_NAME).getText();
+        softly.assertThat(actualData).isEqualTo(EXPECTED_DATA);
+
+        //5. Assert that there are 4 items on the header section are displayed and they have proper texts
+        String actualHomeData = driver.findElement(HOME_BUTTON).getText();
+        softly.assertThat(actualHomeData).isEqualTo(EXPECTED_HOME_DATA);
+
+        boolean isHomeImagePresent = driver.findElement(HOME_BUTTON).isDisplayed();
+        softly.assertThat(isHomeImagePresent).isTrue();
+
+        String actualContactFormData = driver.findElement(CONTACT_FORM_BUTTON).getText();
+        softly.assertThat(actualContactFormData).isEqualTo(EXPECTED_CONTACT_FORM_DATA);
+
+        boolean isContactFormImagePresent = driver.findElement(HOME_BUTTON).isDisplayed();
+        softly.assertThat(isContactFormImagePresent).isTrue();
+
+        String actualServiceData = driver.findElement(SERVICE_BUTTON).getText();
+        softly.assertThat(actualServiceData).isEqualTo(EXPECTED_SERVICE_DATA);
+
+        boolean isServiceImagePresent = driver.findElement(SERVICE_BUTTON).isDisplayed();
+        softly.assertThat(isServiceImagePresent).isTrue();
+
+        String actualMetalsAndColorsData = driver.findElement(METALS_AND_COLORS_BUTTON).getText();
+        softly.assertThat(actualMetalsAndColorsData).isEqualTo(EXPECTED_METALS_AND_COLORS_DATA);
+
+        boolean isMetalsAndColorsImagePresent = driver.findElement(METALS_AND_COLORS_BUTTON).isDisplayed();
+        softly.assertThat(isMetalsAndColorsImagePresent).isTrue();
+
+        //6. Assert that there are 4 images on the Index Page and they are displayed
+        WebElement firstImage = driver.findElement(FIRST_ICON);
+        boolean firstImagePresent = firstImage.isDisplayed();
+        softly.assertThat(firstImagePresent).isTrue();
+
+        boolean isSecondImagePresent = driver.findElement(SECOND_ICON).isDisplayed();
+        softly.assertThat(isSecondImagePresent).isTrue();
+
+        boolean isThirdImagePresent = driver.findElement(THIRD_ICON).isDisplayed();
+        softly.assertThat(isThirdImagePresent).isTrue();
+
+        boolean fourthImagePresent = driver.findElement(FOURTH_ICON).isDisplayed();
+        softly.assertThat(fourthImagePresent).isTrue();
```

Comment:
It will be better get all of this elements as collection, and check `isVisible` through loop

Reply:
I collected web elements to the List and then check its visibility through for each loop.